service/matching/db.go

```diff
@@ -140,8 +140,11 @@ func (db *taskQueueDB) takeOverTaskQueueLocked(
 		}
 		db.ackLevel = response.TaskQueueInfo.AckLevel
 		db.rangeID = response.RangeID + 1
-		db.versioningData = response.TaskQueueInfo.VersioningData
-		return nil
+		_, err = db.getUserDataLocked(ctx)
+		if errors.Is(err, errUserDataNotPresentOnPartition) {
+			return nil
+		}
+		return err
 
 	case *serviceerror.NotFound:
```

Comment:
Hmm.. I didn't think of that. Do you know why we didn't load the versioning data in the original code?

Reply:
Hmm.. might just be oversight.
Yeah, I'll add it.

Reply:
Okay, disregard, I get why now. Adding it. Thanks!

Reply:
I'm not 100% sure about the behavior, but it does seem like it's needed.

you could also do it first, before the `GetTaskQueue` to avoid duplication

Reply:
Oh true

Reply:
Decided against it, I think it's fine to duplicate because it complicated the error handling code